### DISCUSSION TITLE
Improve  code block power {{ something }}

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#14334] [Feature Requests]
Hi,team:

Looks like the block content between `{{` and `}}` just supports variable access. how about 
* formal syntax. `{{ $last['field'] + $last['another_field'] }} `
* basic logics or full javascript:  ` {{ $last['field'] || 'default' }} `  and `{{ if($last['field']) { $last['field_b']} else { $last['field_c']}`

this is very helpful for the transform operators.

**Comment by imiskolee** [comment]
related https://github.com/directus/directus/issues/14330

**Comment by imiskolee** [comment]
I have impl a new feature called `Eval` on the template resolveFn.

```JSON
{
 "a" : "{{$trigger.payload.a}}",
"add" : "{{Eval($trigger.payload.a + $trigger.payload.b)}}",
"more" : "{{Eval($trigger.payload.a + $trigger.payload.b - $trigger.payload.c)}}"
}

```

https://github.com/imiskolee/directus/pull/2/commits/cc039e7a86e8f308fa93ed179854c0531d0e7d2e

**Reply by rijkvanzanten** [reply]
We definitely shouldn't execute any old javascript to implement this, it's way too big of a security risk unfortunately.

**Reply by imiskolee** [reply]
> We definitely shouldn't execute any old javascript to implement this, it's way too big of a security risk unfortunately.

yes, absolutely agreed. but we also need some programming power.

**Comment by imiskolee** [comment]
and I checked the current template engine https://github.com/userpixel/micromustache, it's no features for compute / filters/functions. So  I am not sure can we look another template engine.

**Reply by rijkvanzanten** [reply]
We could look into upgrading that to regular `mustache`, that indeed does light filtering / computing as well

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
